### ISSUE TITLE
[Feature]: Pivot Table Widget

### ISSUE THREAD

**Issue by Nikhil-Nandagopal** [appsmithorg/appsmith#9031]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

A user would like a widget that works as a pivot table to display data and be able to drill down on information

### Why should this be worked on?

It's a community requested feature and we should prioritise it if more users request for it

**Comment by somangshu** [comment]
@Nikhil-Nandagopal could you explain the use case a bit more, Do you mean something like a DB join

**Comment by Nikhil-Nandagopal** [comment]
@somangshu a pivot table is like an interactable table widget where the end user can perform queries on the table, drilldown / rollup data etc.

**Comment by momcilo-appsmith** [comment]
@Nikhil-Nandagopal Is this something like excel's table?

**Comment by Nikhil-Nandagopal** [comment]
@momcilo-appsmith yes exactly. It's fairly complicated so I suggest we hold off till we have a few more requests for this

**Comment by vmulholl** [comment]
Hello, another advantage of pivot tables is simply for the easy of reading through collapsibility and grouping functionality like in Coda tables ([see their grouping functionality](https://help.coda.io/en/articles/773108-grouping-your-table-data)). This avoids table data being repeated row by row, but obviously requires the "values" to be assigned an aggregation calculation (sum, count, average, etc.).

![image](https://user-images.githubusercontent.com/30245917/159691810-5ee90818-d0f8-41a3-92c1-3a4963ab2c39.png)

Why this is useful:
* better UX - gives a cleaner look and feel for the users when looking at larger amounts of data that would typically be grouped (e.g. monthly invoice records, grouped by user/type/category/any dimension, etc.)
* maintains the search functionality from the table widget which is extremely useful

Just wanted to let you know that we would definitely use this feature to improve the user experience in our apps. Thanks for your support!
Vero

**Comment by Nikhil-Nandagopal** [comment]
Thanks for the wonderful information @vmulholl !
